rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,571 @@
+```json
+{ "carNumber": "3800"}
+# or
+"unmodified"
+# or 
+null
+```
+
+### Time
+A [GTFS Time](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#field-types).
+> Time in the HH:MM:SS format (H:MM:SS is also accepted). The time is measured from "noon minus 12h" of the service day (effectively midnight except for days on which daylight savings time changes occur). For times occurring after midnight, enter the time as a value greater than 24:00:00 in HH:MM:SS local time for the day on which the trip schedule begins.  
+> _Example: `14:30:00` for 2:30PM or `25:35:00` for 1:35AM on the next day._
+
+### Trip Key
+For scheduled trips, Glides does not have the same trip ID used by RTR. For added trips, while an ID is provided, it can be the case that an added trip from the Glides perspective is matched to a GTFS trip, in which case RTR will use the GTFS trip ID in publishing the information.
+
+In order to reduce message duplication, a Trip Key is used to identify both added and scheduled trips.
+
+**Added Trip**
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `glidesId` (string): unique value representing this trip for the given `serviceDate`
+```json
+{"serviceDate": "2023-01-20", "glidesId": "ADDED-123"}
+```
+**Scheduled Trip**
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `startLocation` (Location): where the trip is scheduled to start
+- `endLocation` (Location): where the trip is scheduled to end
+- `departureTime` (Time): when the trip is scheduled to depart `startLocation`
+- `arrivalTime` (Time): when the trip is scheduled to arrive at `endLocation`
+
+```json
+{
+  "serviceDate": "2023-01-20",
+  "startLocation": {"gtfsId": "place-mdftf"},
+  "endLocation": {"gtfsId": "place-heath"},
+  "departureTime": "4:47:00",
+  "arrivalTime": "5:34:00"
+}
+```
+*Notes*:
+- The added/scheduled distinction here is internal to Glides, and may not align with what is in other data sources. For example, an Added Trip in Glides may still correspond to a scheduled GTFS trip, and a Scheduled Trip may not appear in GTFS (due to track closures, for example)
+
+## Events
+### Batched
+If multiple events are generated as a part of the same author action, then they can be combined into a single Batched event. This ensures that all of the events are kept together throughout the data pipeline.
+
+Event type: `com.mbta.ctd.glides.batched`
+Fields in the event:
+- `author` (Author, optional): the inspector who performed the events, if it is the same for all the events in the `events` array. 
+- `events` (array of `{"type": string, "data": JSON}`): each element in the array is a CloudEvent including only the `type` and `data` field, and the `data` fields of the individual events do not need to re-include the `author` field if it is the same as the top-level `author`. See [Examples][#examples] for an example.
```

Comment:
Batched events with different authors are not currently possible within Glides and are not expected to be soon, so this possibility is out of scope for the RFC and should not be present here. The `author` should be required in the `batched` event and elided from all sub-events.

Reply:
Batched events without authors are possible, when the schedules are imported.

Reply:
In that case, the event still has an `author`: the user who imported the schedules. The author just isn't associated with a location. (So `author.location` would have to be optional.) ((Which is another reason that if the trip-updated events are done relative to a location, they should include their own location field.))